include/z64camera.h

```diff
@@ -1116,41 +1240,90 @@ typedef struct {
     { flags, CAM_DATA_FLAGS }
 
 typedef struct {
-    /* 0x0 */ s16 idx;
+    /* 0x0 */ s16 index;
+} Special7ReadWriteData; // size = 0x4
+
+typedef struct {
+    /* 0x0 */ Special7ReadWriteData rwData;
 } Special7; // size = 0x4
 
+typedef struct {
+    /* 0x0 */ s16 interfaceFlags;
+} Special6ReadOnlyData; // size = 0x4
+
 typedef struct {
     /* 0x0 */ f32 initalPlayerY;
     /* 0x4 */ s16 animTimer;
-} Special6Anim; // size = 0x8
+} Special6ReadWriteData; // size = 0x8
 
 typedef struct {
-    /* 0x0 */ s16 interfaceFlags;
-    /* 0x4 */ Special6Anim anim;
+    /* 0x0 */ Special6ReadOnlyData roData;
+    /* 0x4 */ Special6ReadWriteData rwData;
 } Special6; // size = 0xC
 
-typedef struct {
-    /* 0x0 */ s16 targetYaw;
-} Special9Anim; // size = 0x2
-
 typedef struct {
     /* 0x0 */ f32 yOffset;
     /* 0x4 */ f32 unk_04;
     /* 0x8 */ s16 interfaceFlags;
     /* 0xA */ s16 unk_0A;
-    /* 0xC */ Special9Anim anim;
-} Special9Params; // size = 0x10
+} Special9ReadOnlyData; // size = 0xC
 
 typedef struct {
-    /* 0x0 */ DoorParams doorParams;
-    /* 0xC */ Special9Params params;
+    /* 0x0 */ s16 targetYaw;
+} Special9ReadWriteData; // size = 0x4
+
+typedef struct {
+    /* 0x00 */ DoorParams doorParams;
+    /* 0x0C */ Special9ReadOnlyData roData;
+    /* 0x18 */ Special9ReadWriteData rwData;
 } Special9; // size = 0x1C
 
 #define CAM_FUNCDATA_SPEC9(yOffset, fov, flags) \
     { yOffset, CAM_DATA_Y_OFFSET }, \
     { fov, CAM_DATA_FOV }, \
     { flags, CAM_DATA_FLAGS }
 
+typedef union {
+    Normal1 norm1;
+    Normal2 norm2;
+    Normal3 norm3;
+    Parallel1 para1;
+    Jump1 jump1;
+    Jump2 jump2;
+    Jump3 jump3;
+    Battle1 batt1;
+    Battle4 batt4;
+    KeepOn0 keep0;
+    KeepOn1 keep1;
+    KeepOn3 keep3;
+    KeepOn4 keep4;
+    Fixed1 fixd1;
+    Fixed2 fixd2;
+    Fixed3 fixd3;
+    Fixed4 fixd4;
+    Subj3 subj3;
+    Subj4 subj4;
+    Data4 data4;
+    Unique0 uniq0;
+    Unique1 uniq1;
+    Unique2 uniq2;
+    Unique3 uniq3;
+    Unique6 uniq6;
+    Unique7 uniq7;
+    Unique9 uniq9;
+    Demo1 demo1;
+    Demo3 demo3;
+    Demo6 demo6;
+    Demo9 demo9;
+    Special0 spec0;
+    Special4 spec4;
+    Special5 spec5;
+    Special6 spec6;
+    Special7 spec7;
+    Special9 spec9;
+    DoorParams doorParams;
```

Comment:
The reason DoorParams is there is because of `Camera_ChangeDoorCam(`. Separating it out from the union would break the struct, and DoorParams is intended to be there as part of `Unique3`, `Unique0`, and `Special9`. So because it's unspecified which of the 3 it's being allocated to, it's its own separate part of the union.

Reply:
Ah I see where you're coming from with this. I'm not sure if it's worth going this extra step because it's quite messy as you mentioned